test/test_list.cpp

```diff
@@ -738,10 +738,11 @@ TEST(List_Nested_InMixed)
     dict->insert("Dict", Mixed());
     CHECK_THROW_ANY_GET_MESSAGE(dict2->insert("Five", 5), message); // This dictionary ceased to be
     CHECK_EQUAL(message, "This collection is no more");
-    // Try to insert a new dictionary. The old dict2 shoulg still be stale
-    dict->insert_collection("Dict", CollectionType::Dictionary);
-    CHECK_THROW_ANY_GET_MESSAGE(dict2->insert("Five", 5), message); // This dictionary ceased to be
-    CHECK_EQUAL(message, "This collection is no more");
+    // Try to insert a new dictionary. The old dict2 should still be stale
+    // Well - we can't be sure of that. But it would not be critical - it is still a dictionary
+    // dict->insert_collection("Dict", CollectionType::Dictionary);
```

Comment:
What happens to the values that are stored in the dictionary?  I guess there might be a case where the dictionary is not stale and all the values are gone? Would this be a problem? Probably not, since this should have happened because set_collection has been called.?

Reply:
The accessor is just updated to this new empty directory.